docs/proposals/dynamic-app-parameters.md

```diff
@@ -0,0 +1,181 @@
+
+When access to the resource is forbidden by Kubernetes, the resource will not exist in the cluster cache. In this case, a generic “Resource not found” error should be returned, without disclosing any other error details to the user. Details can be logged instead.
+
+### Unknown kind or group
+
+When the user references a kind or group that does not exist on the destination cluster (possibly a typo in the configuration, etc), a detailed error message may be displayed to the user, since it is assumed that the user would have access to the resource.
+
+### Referenced resource does not exist
+
+When the referenced resource does not exist, and path is not set, it will set the parameter’s value to false. If a path is given, an error is returned to the user that the referenced resource does not exist.
+
+### Field in referenced resource does not exist
+
+When there is a field reference given as path, but this path cannot be found in the referenced resource, a specific error message (e.g. path `.data.foo` does not exist) may be returned to the user, since it is assumed that the user has access to the resource.
+
+## Security implications
+
+### Preventing reading arbitrary cluster resources
+
+Dynamic parameters must not be allowed to read arbitrary resources on the cluster. Instead, there must be some access control mechanisms.
+
+Access control for retrieving values from clusters will be enforced by restrictions in the `AppProject`, some of which already are in place, and will be using the following principles:
+
+* If access to the resource would be allowed through existing `namespaceResourceWhitelist`, `namespaceResourceBlacklist`, `clusterResourceWhitelist` and `clusterResourceBlacklist` configuration, then a dynamic parameter may access the resource as well.
+* Additionally, new access controls specifically for allowing read-only access to dynamic parameters should be introduced. In addition to referencing accessable resources through `kind`, `group` and (possibly) `namespace` attributes, it should optionally be more refinable by a specific resource `name`. The fields should be called `namespaceReadOnlyAllowlist` and `clusterReadOnlyAllowList` respectively, and not be implicitly bound by the field name to the dynamic parameter feature for future applications. There's no need (yet) for a deny list in this scenario.
+
+### Transparency of dynamic parameters' values
+
+It should be either documented that dynamic parameters should not refer to confidential data stored in a cluster (e.g. a Secret), or we need to take care not to leak any of the values back to a user who has e.g. `get` or `sync` access for the Application. Leakage could occur anywhere, in error messages, logs, etc.
+
+We should keep in mind that even with dynamic parameters, the values will be part of the command line when `helm template` is executed.
+
+Therefore, the recommendation is to make it clear to not use this feature for confidential items.
```

Comment:
Block access to Secrets unless someone manually overrides?

Reply:
Good thinking. So this would be a knob somewhere, that the admin (or the user?) has to explicitly enable.